backend/src/main/java/com/ambacam/service/ActionService.java

```diff
@@ -0,0 +1,115 @@
+package com.ambacam.service;
+
+import com.ambacam.exception.ResourceBadRequestException;
+import com.ambacam.exception.ResourceNotFoundException;
+import com.ambacam.model.Action;
+import com.ambacam.repository.ActionRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@Transactional(rollbackFor = Exception.class)
+public class ActionService {
+	@Autowired
+	private ActionRepository actionRepository;
+
+	/**
+	 * Create an action
+	 * 
+	 * @param action The new action to create
+	 * 
+	 * @return The new action created
+	 * @throws ResourceBadRequestException if an action with the name already exist
+	 */
+	public Action create(Action action) {
+
+		checkConsistency(action);
+
+		action.setId(null);
+
+		return actionRepository.save(action);
+	}
+
+	/**
+	 * Get an action
+	 * 
+	 * @param id The id of the Action you want to get
+	 * 
+	 * @return The action found
+	 * 
+	 * @throws ResourceNotFoundException if the action does not exist
+	 */
+	public Action get(Long id) {
+		return findAction(id);
+	}
+
+	/**
+	 * List all actions
+	 * 
+	 * @return The list of action stored
+	 */
+	public List<Action> list() {
+
+		return actionRepository.findAll();
+	}
+
+	/**
+	 * Update an action
+	 * 
+	 * @param id The id of the action to update
+	 * 
+	 * @param action The new action modifications
+	 * 
+	 * @return The action updated
+	 * 
+	 * @throws ResourceNotFoundException if the action is not found
+	 * @throws ResourceBadRequestException if an action with the name already exist
+	 */
+	public Action update(Long id, Action action) {
```

Comment:
please rename action to update

Reply:
I though that we discuss about that in a previous pull request. no ?

Reply:
Me too, but I thought the name in function update should be just update.

Okay, for now, let's have it as you did. We will follow a new rule:

In the function create:
parameter name: name of the model (e.g: action)
object collected from the database: found

I update everything in my ticket so that it gets updated after merging.